### PR TITLE
Improve `EditorTable` performace

### PR DESCRIPTION
**Pull request by EVAST9919**
092856347a10893e76758172093900768ca1a632 removes `GridContainer` from the input subtree since all the input is being handled by the background.

The rest of pr: implementation of `EditorTableBackground` component, which mimics behavior of a vertical `FillFlowContainer` full of clickable items in an efficient way. Tests are included.

Map example with a huge "control point" and "verify" tables: https://osu.ppy.sh/beatmapsets/1238185#mania/2574372

|master|092856347a10893e76758172093900768ca1a632|full pr|
|---|---|---|
|master|input|pr|

### CODE REVIEW

**Comment by peppy** [comment]
the hell is this diff? is this considered a code quality improvement from your end?

**Comment by EVAST9919** [comment]
> the hell is this diff? is this considered a code quality improvement from your end?

Most of it is a new component + tests. The rest is moving objects handling outside of the background to the table iself, which I would consider a better practice. Or you mean it would be better to make a separate pr for each change?

File: osu.Game/Screens/Edit/EditorTableBackground.cs [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+                hoveredIndexBindable.Value = getItemIndexAtMousePosition();
+        }
+
+        protected override bool OnHover(HoverEvent e) => true;
+
+        protected override void OnHoverLost(HoverLostEvent e)
+        {
+            base.OnHoverLost(e);
+            hoveredIndexBindable.Value = -1;
+        }
```
Comment (bdach) [inline osu.Game/Screens/Edit/EditorTableBackground.cs:95]: No. I am not having the "background" of a table handle input.

File: osu.Game/Screens/Edit/EditorTableBackground.cs [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+                if (newHovered.State == RowState.None)
+                    newHovered.State = RowState.Hovered;
+            }
+            else
+                Add(new EditorTableBackgroundRow(hoveredIndex));
+
+            playHoverSample();
+        }
+
+        public EditorTableBackgroundRow Select(int itemIndex)
```
Comment (bdach) [inline osu.Game/Screens/Edit/EditorTableBackground.cs:128]: I am not having the "row background" of a table having a separate selection method from the actual row. It's the row. The user is performing the operation of selecting the _row_. It is the _**row**_ you're selecting.

File: osu.Game/Screens/Edit/EditorTable.cs [hunk 3]
```diff
@@ -2,87 +2,111 @@
...
         private const float horizontal_inset = 20;
 
-        protected const float ROW_HEIGHT = 25;
-
         public const int TEXT_SIZE = 14;
 
-        protected readonly FillFlowContainer<RowBackground> BackgroundFlow;
+        private readonly List<T> items = new List<T>();
+
+        public virtual void SetNewItems(IEnumerable<T> newItems)
```
Comment (bdach) [inline osu.Game/Screens/Edit/EditorTable.cs:28]: Why?

**Review by bdach — CHANGES_REQUESTED** [review]
Structurally this is unacceptable

**Pull request closed without merging**